src/keepers/HarvestResolver.sol

```diff
@@ -248,27 +258,49 @@ contract HarvestResolver is IResolver, Ownable {
             uint256 minimumAmountOut = currentValue - ytp.expectedValue;
             minimumAmountOut = minimumAmountOut - (minimumAmountOut * h.slippageBps) / SLIPPAGE_PRECISION;
 
-            // If vault has rewards ot be collected
-            if (h.rewardToken != address(0)) {
-              uint256 expectedExchange;
-              // alUSD route
+            uint256 expectedExchange;
+
+            // If vault has rewards to be collected
+            if (h.reward == vspRewardToken) {
+              // alUSD route for vesper swap
               if (h.alchemist == usdAlchemistAddress) {
                 if (h.yieldToken == vaDAI) {
                   (address[] memory tokens, uint256[] memory amounts) = IVesperRewards(0x35864296944119F72AA1B468e13449222f3f0E67).claimable(usdAlchemistAddress);
-                  expectedExchange = _getExpectedExchange(uniswapFactory, h.rewardToken, wethAddress, uint24(3000), dai, uint24(3000), amounts[0]);
+                  expectedExchange = _getExpectedExchange(uniswapFactory, h.reward, wethAddress, uint24(3000), dai, uint24(3000), amounts[0]);
                 } else if (h.yieldToken == vaUSDC) {
                   (address[] memory tokens, uint256[] memory amounts) = IVesperRewards(0x2F59B0F98A08E733C66dFB42Bd8E366dC2cfedA6).claimable(usdAlchemistAddress);
-                  expectedExchange = _getExpectedExchange(uniswapFactory, h.rewardToken, wethAddress, uint24(3000), dai, uint24(3000), amounts[0]);
+                  expectedExchange = _getExpectedExchange(uniswapFactory, h.reward, wethAddress, uint24(3000), dai, uint24(3000), amounts[0]);
                 }
-              // alETH route
+              // alETH route for vesper swap
               } else if (h.alchemist == ethAlchemistAddress) {
                 (address[] memory tokens, uint256[] memory amounts) = IVesperRewards(0x51EEf73abf5d4AC5F41De131591ed82c27a7Be3D).claimable(ethAlchemistAddress);
-                expectedExchange = _getExpectedExchange(uniswapFactory, h.rewardToken, wethAddress, uint24(3000), address(0), uint24(0), amounts[0]);
+                expectedExchange = _getExpectedExchange(uniswapFactory, h.reward, wethAddress, uint24(3000), address(0), uint24(0), amounts[0]);
               }
               return (
                 true,
                 abi.encodeWithSelector(IAlchemixHarvester.harvest.selector, h.alchemist, yieldToken, minimumAmountOut, expectedExchange * 9900 / 10000)
               );
+            }
+
+            // If reward is not the address of a token then it is the address of a reward collector.
+            // We can assume that this is optimism and handle rewards accordingly.
+            if (h.reward != address(0)) {
+              address[] memory token = new address[](1);
+              token[0] = h.aaveToken;
+              uint256 claimable = IRewardsController(0x929EC64c34a17401F460460D4B9390518E5B473e).getUserRewards(token, yieldToken, IRewardCollector(h.reward).rewardToken());
```

Comment:
set all these addresses to constant variables on the contract so we know what they are